app/main.py

```diff
@@ -86,121 +86,59 @@
 }
 
 
-def battle(knightsConfig):
-    # BATTLE PREPARATIONS:
+def put_on(personage: dict) -> dict:
+    our_personage = personage.copy()
+    our_personage["protection"] = 0
+    for armour in personage["armour"]:
+        our_personage["protection"] += armour["protection"]
 
-    # lancelot
-    lancelot = knightsConfig["lancelot"]
+    our_personage["power"] += our_personage["weapon"]["power"]
 
-    # apply armour
-    lancelot["protection"] = 0
-    for a in lancelot["armour"]:
-        lancelot["protection"] += a["protection"]
+    if our_personage["potion"] is not None:
+        our_p = our_personage["potion"]
+        if "power" in our_p["effect"]:
+            our_personage["power"] += our_p["effect"]["power"]
 
-    # apply weapon
-    lancelot["power"] += lancelot["weapon"]["power"]
+        if "protection" in our_p["effect"]:
+            our_personage[
+                "protection"
+            ] += our_p["effect"]["protection"]
 
-    # apply potion if exist
-    if lancelot["potion"] is not None:
-        if "power" in lancelot["potion"]["effect"]:
-            lancelot["power"] += lancelot["potion"]["effect"]["power"]
+        if "hp" in our_p["effect"]:
+            our_personage["hp"] += our_personage["potion"]["effect"]["hp"]
 
-        if "protection" in lancelot["potion"]["effect"]:
-            lancelot["protection"] += lancelot["potion"]["effect"]["protection"]
+    return our_personage
 
-        if "hp" in lancelot["potion"]["effect"]:
-            lancelot["hp"] += lancelot["potion"]["effect"]["hp"]
 
-    # arthur
-    arthur = knightsConfig["arthur"]
+def vs(pers1: dict, pers2: dict) -> dict:
+    pers1["hp"] -= pers2["power"] - pers1["protection"]
```

Comment:
The calculation `pers1['hp'] -= pers2['power'] - pers1['protection']` can result in negative damage if `pers1['protection']` is greater than `pers2['power']`. Consider adding a check to ensure that damage is not negative.